control/autoware_collision_detector/src/node.cpp

```diff
@@ -0,0 +1,523 @@
+
+    // Calculate object distance from base_link
+    const double object_distance = transformed_position.head<2>().norm();
+    const bool is_within_range = (object_distance <= node_param_.nearby_filter_radius);
+
+    // Determine if the object should be filtered based on its classification
+    bool should_be_filtered_class = shouldBeFiltered(object.classification.front().label);
+
+    const bool is_within_range_and_filtering_class = is_within_range && should_be_filtered_class;
+
+    // If the object is not within range or not a class to be filtered, add it directly
+    if (!is_within_range_and_filtering_class) {
+      filtered_objects.objects.push_back(object);
+
+      // Update observed_objects_
+      auto observed_it = std::find_if(
+        observed_objects_.begin(), observed_objects_.end(),
+        [&object](const auto & observed_object) {
+          return observed_object.object_id == object.object_id;
+        });
+      if (observed_it != observed_objects_.end()) {
+        observed_it->timestamp = current_object_time;
+      } else {
+        observed_objects_.push_back({object.object_id, current_object_time});
+      }
+
+      continue;
+    }
+
+    // Check if the object exists in ignored_objects_
+    auto ignored_it = std::find_if(
+      ignored_objects_.begin(), ignored_objects_.end(), [&object](const auto & ignored_object) {
+        return ignored_object.object_id == object.object_id;
+      });
+    const bool was_ignored = (ignored_it != ignored_objects_.end());
+
+    // If the object was ignored and is still within the ignore period, continue filtering
+    if (
+      was_ignored && (current_object_time - ignored_it->timestamp) <
+                       rclcpp::Duration::from_seconds(node_param_.keep_ignoring_time)) {
+      // Check if the object exists in observed_objects_
+      auto observed_it = std::find_if(
+        observed_objects_.begin(), observed_objects_.end(),
+        [&object](const auto & observed_object) {
+          return observed_object.object_id == object.object_id;
+        });
+      const bool was_observed = (observed_it != observed_objects_.end());
+      if (was_observed) {
+        observed_it->timestamp = current_object_time;
+      } else {
+        // Add as a newly observed object and to the ignore list
+        observed_objects_.push_back({object.object_id, current_object_time});
+      }
+      continue;
+    }
+
+    // Check if the object exists in observed_objects_
+    auto observed_it = std::find_if(
+      observed_objects_.begin(), observed_objects_.end(), [&object](const auto & observed_object) {
+        return observed_object.object_id == object.object_id;
+      });
+    const bool was_observed = (observed_it != observed_objects_.end());
+
+    if (was_observed) {
+      observed_it->timestamp = current_object_time;
+      // Add without filtering
+      filtered_objects.objects.push_back(object);
```

Comment:
I am confused about what "filtering" means here. The comment says we are not filtering, but we are adding the object to a "filtered" list?

Reply:
@ambroise-arm 
Thank you for your comments. I realized that my use of "filter" was inappropriate. 
The term `filtered_objects` represents a list of objects that meet the specified conditions, but I mistakenly used `filtering` to describe the exclusion of objects, which caused confusion. Therefore, I plan to unify the terminology by explicitly indicating that `filtered_objects` is a list of objects meeting the conditions and avoid using `filter` to describe exclusion. If you agree, I will proceed with the changes based on this approach.

Reply:
Sounds good

Reply:
Definition changed in 9b89d41266d4f0ae620843e6794376e30d496032